### PR TITLE
Feature/client implementation

### PR DESCRIPTION
**Pull request by AlexandrZagorskiy**
- Code refactoring. Each operation has the same interface for its implementation, which allows you to decompose the code. It is necessary for the implementation of unit tests.
- The tests have been transferred to the Pytest framework.
- Added development tools. The pre-commit config has been added, and CI has also been configured.

### CODE REVIEW

File: .github/workflows/pull-request-CI.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        # You may pin to the exact commit or the version.
+        # uses: GsActions/commit-message-checker@9d8708beab99f811c5fe3a4f98acc4b2f2ba8496
+        uses: GsActions/commit-message-checker@v1
+        with:
+          # A regex pattern to check if a commit message is valid.
+          pattern: "((feat|fix|docs|style|refactor|test|build|ci|perf)+(\\([a-z]+(.)+\\))?\\: [a-z]+(.)+)|(Merge(.)*)|(Review fixes(.)*)$"
+          # Expression flags change how the expression is interpreted.
+          flags: # optional, default is gm
+          # A error message which will be returned in case of an error.
+          error: "One of commit messages has an incorrect title. Please read the documentation: docs/dev/pr.md"
```
Comment (MikhailSadovsky) [inline .github/workflows/pull-request-CI.yml:25]: We haven't link `docs/dev/pr.md` in repo. Could you please refer to existing link?

File: src/sc_client/client/__init__.py [hunk 2]
```diff
@@ -5,9 +5,19 @@
 """
```
Comment (MikhailSadovsky) [inline src/sc_client/client/__init__.py:5]: Question regarding project structure: is it ok to have sc_client/client folders inside py-sc-client project?

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,9 @@
+[tox]
```
Comment (MikhailSadovsky) [inline tox.ini:1]: Could you please update README with info how to install lib

**Review by MikhailSadovsky — CHANGES_REQUESTED** [review]

File: .github/workflows/push-CI.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        cache: 'pip'
+        cache-dependency-path: 'requirements-dev.txt'
+
+    - name: Checking with isort
+      run: |
+        isort -c src
+
+    - name: Checking with black
+      run: |
+        black --diff --color --line-length 120 src
```
Comment (Vikort) [inline .github/workflows/push-CI.yml]: with `diff` flag black return 0 code even if files need to be reformatted, use `check` argument instead

File: src/sc_client/constants/common.py [hunk 5]
```diff
@@ -18,6 +18,20 @@ class RequestTypes:
...
+    GET_LINK_CONTENT = auto()
+    GET_LINKS_BY_CONTENT = auto()
+    KEYNODES = auto()
+    SEARCH_TEMPLATE = auto()
+    GENERATE_TEMPLATE = auto()
+    EVENTS_CREATE = auto()
+    EVENTS_DESTROY = auto()
+
+
 BINARY = "binary"
```
Comment (Vikort) [inline src/sc_client/constants/common.py:35]: Why this constants not in `Enum` classes? Why some classes with constants are Enum and others are not?

File: .github/workflows/run-tests.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      with:
+        python-version: ${{ matrix.python }}
+        cache: 'pip'
+        cache-dependency-path: 'requirements-dev.txt'
+    - name: update pip
+      run: |
+        pip install -U wheel
+        pip install -U setuptools
+        python -m pip install -U pip
+    - run: pip install tox
```
Comment (Vikort) [inline .github/workflows/run-tests.yml:30]: Why install `tox` if it installed from `requirements-dev.txt`?

File: src/sc_client/client/__init__.py (hunk 2 above)
Comment (AlexandrZagorskiy) [reply src/sc_client/client/__init__.py:5]: Why not ? I didn't see any problems with such naming and structure. Also it looks good for import:
`from sc_client import client`
I admit that project can be restructured, there can be some shortcomings, the reason is just a lack of experience

File: src/sc_client/constants/common.py (hunk 5 above)
Comment (AlexandrZagorskiy) [reply src/sc_client/constants/common.py:35]: I used Enum mostly for parts of code when I need to use both keys and values. I think there is no need to move every constant to class and inherit it from Enum.

**Review by Vikort — APPROVED** [review]

File: docs/dev/git-workflow.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+## Installation
```
Comment (MikhailSadovsky) [inline docs/dev/git-workflow.md]: Please move to separate installation doc file

File: docs/dev/git-workflow.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+pre-commit install
+```
+
+Run tests:
+```sh
+tox -e py38
+```
+
+### Workflow
+This project uses [Git-Flow](https://www.gitkraken.com/learn/git/git-flow),
```
Comment (MikhailSadovsky) [inline docs/dev/git-workflow.md:18]: Please add info about fork creation

**Review by MikhailSadovsky — CHANGES_REQUESTED** [review]

File: .github/workflows/run-tests.yml [hunk 9]
```diff
@@ -0,0 +1,32 @@
+name: Tests
```
Comment (MikhailSadovsky) [inline .github/workflows/run-tests.yml:1]: Please fix CI issues and add CI badge to README

**Review by MikhailSadovsky — CHANGES_REQUESTED** [review]

**Review by MikhailSadovsky — APPROVED** [review]

**Pull request merged**